src/averageAges.js

```diff
@@ -55,7 +75,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const motherSons = people.filter((person) => onlyWithSon
+    ? person.sex === 'm' && people.find((mother) =>
+      person.mother === mother.name)
+    : people.find((mother) => person.mother === mother.name));
```

Comment:
`people.find((mother) => person.mother === mother.name)` this expression also needs a variable